src/vs/workbench/contrib/tasks/common/taskService.ts

```diff
@@ -19,6 +19,7 @@ export { ITaskSummary, Task, ITaskTerminateResponse as TaskTerminateResponse };
 
 export const CustomExecutionSupportedContext = new RawContextKey<boolean>('customExecutionSupported', true, nls.localize('tasks.customExecutionSupported', "Whether CustomExecution tasks are supported. Consider using in the when clause of a \'taskDefinition\' contribution."));
 export const ShellExecutionSupportedContext = new RawContextKey<boolean>('shellExecutionSupported', false, nls.localize('tasks.shellExecutionSupported', "Whether ShellExecution tasks are supported. Consider using in the when clause of a \'taskDefinition\' contribution."));
+export const TaskCommandsRegistered = new RawContextKey<boolean>('taskCommandsRegistered', false, nls.localize('tasks.taskCommandsRegistered', "Whether the task commands have been registered yet"));
```

Comment:
Instead of a context key could the actions be registered early? This context key means that hitting it early in startup won't do anything, unlike most other commands

Reply:
that seems like it would be involved and too risky for this close to endgame IMO

Reply:
which I can no longer reproduce with the recent changes

Reply:
I agree there isn't enough time to look at this now, but the ideal UX we should shoot for is ctrl+shift+b on start up queues the build task request until tasks is ready